cmd/alertmanager/main.go

```diff
@@ -172,6 +169,12 @@ func main() {
 			level.Error(logger).Log("msg", "Unable to initialize gossip mesh", "err", err)
 			os.Exit(1)
 		}
+		ctx, cancel := context.WithTimeout(context.Background(), *settleTimeout)
+		defer func() {
+			cancel()
+			peer.Leave(10 * time.Second)
+		}()
+		go peer.Settle(ctx, *pushPullInterval / 2)
```

Comment:
`*pushPullInterval / 2` means that by default the interval is 30s while at the same time the context time-out is 60s. In practice the Settle() function always terminates on context time-out since it waits for 4 intervals.

Reply:
You are right, I maybe let's use `gossipItnerval * 10` instead. I know it's a magic value, but it should mostly work.